trackers/node-tracker/src/tracker.ts

```diff
@@ -48,31 +49,33 @@ export interface Tracker extends TrackerCore {
   setNetworkUserId: (userId: string) => void;
 }
 
-/**
- * Snowplow Node.js Tracker
- *
- * @param string - or array emitters The emitter or emitters to which events will be sent
- * @param string - namespace The namespace of the tracker
- * @param string - appId The application ID
- * @param boolean - encodeBase64 Whether unstructured events and custom contexts should be base 64 encoded
- */
+interface NodejsTrackerConfiguration {
+  /* The namespace of the tracker */
+  namespace: string;
+  /* The application ID */
+  appId: string;
+  /* Whether unstructured events and custom contexts should be base64 encoded. */
+  encodeBase64: boolean;
+}
+
+type CustomEmitter = {
+  /* Function returning custom Emitter or Emitter[] to be used. If set, other options are irrelevant */
+  customEmitter: () => Emitter | Array<Emitter>;
+};
+
+type NodejsEmitterConfiguration = CustomEmitter | GotEmitterConfiguration;
+
 export function tracker(
-  emitters: Emitter | Array<Emitter>,
-  namespace: string,
-  appId: string,
-  encodeBase64: boolean
+  trackerConfiguration: NodejsTrackerConfiguration,
+  emitterConfiguration: NodejsEmitterConfiguration
```

Comment:
Wouldn't it be better to also accept an array of `NodejsEmitterConfiguration` here instead of only through the `customEmitter` function? Because now I think users can't easily make multiple got emitter instances like they could before – if they want to have multiple emitters, they need to implement custom ones.

Reply:
But you said that the `gotEmitter` function is no longer exported. If it is, then I have no problem with this. I was just worried that if people want to have multiple emitters, they need to implement their own emitter from scratch.

Reply:
Yes the `gotEmitter` function is not exported now _(I left the comment `/* For the client, this would be any kind of function they want */` to make it clear that this would be custom made)._

> if people want to have multiple emitters, they need to implement their own emitter from scratch

Yes that is the case, but we provide the interface and point them to our own implementation. They can even copy the code and use it exactly as stated above. I considered this a fair change to remove. We can provide this convenience but in my view it adds more complication to our API rather than a clear benefit.

I might be wrong since I might be missing past context, but is this a commonly used feature ?

Reply:
But the complication from our side is like adding a couple of extra lines of code while for the user it's adding maybe 100 lines so it's quite disproportionate.

I don't think the feature is used much but I also don't have any insight into how people use the tracker. Also our other trackers don't have the same feature. I am just arguing for it because some users of v3 might have adopted it and will be sad when they upgrade to v4. But I don't have a strong opinion either way.

Reply:
This is one of those features which when its used, its a strong requirement, but it is used quite rarely. Initialising multiple trackers is miserable, so I think keeping a way of creating two "core" emitters is a feature we need to leave in.

We could make it `emitterConfiguration: EmitterConfiguration | Array<EmitterConfiguration>`. It doesn't make our side much worse, but it gives great flexibility for the end user. 

In reality no one is going to build their own emitter, it's probably a risky venture for most and they'd prefer us to deal with the issues and bugs. So if anything, putting the work into this over customEmitter is probably more important (although leaving both in is good too for flexibility).

+1 for not exporting `gotEmitter` thought and moving to `EmitterConfiguration`s.